client/ayon_core/pipeline/context_tools.py

```diff
@@ -579,3 +586,39 @@ def get_process_id():
     if _process_id is None:
         _process_id = str(uuid.uuid4())
     return _process_id
+
+
+def version_up_current_workfile():
+    """Function to increment and save workfile
+    """
+    host = registered_host()
+    project_name = get_current_project_name()
+    folder_path = get_current_folder_path()
+    task_name = get_current_task_name()
+    host_name = get_current_host_name()
+
+    template_key = get_workfile_template_key_from_context(
+        project_name,
+        folder_path,
+        task_name,
+        host_name,
+    )
+    anatomy = Anatomy(project_name)
+
+    data = get_template_data_with_names(
+        project_name, folder_path, task_name, host_name
+    )
+    data["root"] = anatomy.roots
```

Comment:
This seems overly specific 'hack' that is likely copied from elsewhere where this code is based on. Copying that logic here isn't necessarily wrong but does point to something that we might want to expose in a more sensible manner for others to use.

This begs the question - why is it this involved or complex from an API standpoint to define what the next potential workfile name should be? Shouldn't we expose that in a better way for easier re-use (also if someone from another addon may want to figure that out?)

For example, if there were a `get_next_workfile_version_filepath(context)` function then this would be more easily accessible for re-use without it also automatically saving the file. I would want to avoid us needing to copy this exact logic around to define a workfile filename from anywhere.

May not have to be this PR - but just wanted to remark that we should definitely improve the API access here.